### PR TITLE
fix: Add egress rule update support

### PR DESCRIPTION
**Pull request by renebarbosafl**
## Description

This PR complements https://github.com/terraform-aws-modules/terraform-aws-route53/pull/106 by adding a new variable, allowing the submodule to update egress rules. This PR also adds support for setting static IP address on the resolver endpoint instances.

## Motivation and Context

The previous pull request was hardcoding an egress rule allowing All the traffic. Users could want to update these rules as their needs.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by renebarbosafl** [comment]
Hi @antonbabenko, Is there any chance we could get this approved and merged? 
Thank you very much!

**Comment by renebarbosafl** [comment]
Commenting to remove stale mark.

File: modules/resolver-endpoints/main.tf [hunk 1]
```diff
@@ -12,10 +12,11 @@ resource "aws_route53_resolver_endpoint" "this" {
   security_group_ids     = local.security_group_ids
 
   dynamic "ip_address" {
-    for_each = var.subnet_ids
+    for_each = var.ip_address
 
     content {
-      subnet_id = ip_address.value
+      ip        = ip_address.value.ip
```
Comment (antonbabenko) [inline modules/resolver-endpoints/main.tf:18]:
```suggestion
      ip        = lookup(ip_address.value, "ip", null)
```

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_resolver_endpoint#ip - here it says that "ip" is optional.

File: modules/resolver-endpoints/variables.tf [hunk 2]
```diff
@@ -28,10 +28,13 @@ variable "type" {
   default     = "IPV4"
 }
 
-variable "subnet_ids" {
-  description = "A list of subnets where Route53 resolver endpoints will be deployed"
-  type        = list(string)
-  default     = []
+variable "ip_address" {
+  description = "A list of IP addresses and subnets where Route53 resolver endpoints will be deployed"
+  type = list(object({
```
Comment (antonbabenko) [inline modules/resolver-endpoints/variables.tf:33]:
```suggestion
  type = list(map(string)) # or list(any)
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Let's make this change not as a breaking change.

Add support for `ip_address` variable like now; keep `subnet_ids` as before.

In `for_each` try to use `ip_address` (if not empty) or `subnet_ids` converted in the same structure as `ip_address`.

Make sure to show the usage of both variables in examples.

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 4.0.1 :tada:

**Pull request merged**